lib/UserBackend.php

```diff
@@ -400,4 +402,24 @@ public function loginName2UserName($loginName) {
 	public function getBackendName() {
 		return 'Guests';
 	}
+
+	public function getRealUID(string $uid): string {
+		$qb = $this->dbConn->getQueryBuilder();
+		$qb->select('uid')
+			->from('guests_users')
+			->where(
+				$qb->expr()->eq(
+					'uid_lower', $qb->createNamedParameter(mb_strtolower($uid))
+				)
+			);
+		$result = $qb->execute();
+		$row = $result->fetch();
+		$result->closeCursor();
+
+		if (!$row) {
+			throw new \RuntimeException($uid . ' does not exist');
```

Comment:
Not annotated in the interface but this is the same behavior that the database backend has https://github.com/nextcloud/server/blob/734c62bee0f46652285f782761317889d1ef5c7d/lib/private/User/Database.php#L485